jvm-agent/src/main/org/jetbrains/lincheck/jvm/agent/TraceAgentParameters.kt

```diff
@@ -23,24 +29,52 @@ object TraceAgentParameters {
     var traceDumpFilePath: String? = null
 
     @JvmStatic
-    private var restOfArgs: MutableList<String> = mutableListOf()
+    private val namedArgs: MutableMap<String, String?> = mutableMapOf()
 
     @JvmStatic
-    fun parseArgs(args: String?) {
+    fun parseArgs(args: String?, validAdditionalArgs: List<String>) {
         if (args == null) {
             error("Please provide class and method names as arguments")
         }
-        val actualArguments = splitArgs(args)
-        classUnderTraceDebugging = actualArguments.getOrNull(0) ?: error("Class name was not provided")
-        methodUnderTraceDebugging = actualArguments.getOrNull(1) ?: error("Method name was not provided")
-        traceDumpFilePath = actualArguments.getOrNull(2)
-        if (actualArguments.size > 3) {
-            restOfArgs.addAll(actualArguments.subList(3, actualArguments.size))
+        // Try to parse new-style
+        val kvArguments = parseKVArgs(args)
+        if (kvArguments == null) {
+            Logger.warn { "Looks like old-style arguments found, consider migrate to key-value arguments" }
```

Comment:
Do we need to support old-style? 
Can we just update all usages on plugin side and migrate in one step?